docs/tutorials/Remark.adoc

```diff
@@ -0,0 +1,396 @@
+
+=== Make use of `Remark`
+
+Let's change `RemarkCommand` and `RemarkCommandParser` to use the new `Remark` class instead of plain `String`.
+These should be relatively simple changes.
+
+== Add a placeholder element for remark to the UI
+
+Without getting too deep into `fxml`, let's go on a 5 minute adventure to get some placeholder text to show up for each person.
+
+Simply add
+[source, java]
+PersonCard.java
+```
+@FXML
+private Label remark;
+```
+
+to link:https://github.com/j-lum/addressbook-level35/commit/06f9897d7088d22689de478c94612f8bcbff2964#diff-0c6b6abcfac8c205e075294f25e851fe[`seedu.address.ui.PersonCard`].
+`@FXML` is an annotation that marks a private or protected field and makes it accessible to FXML.
+It might sound like Greek to you right now, don't worry -- we will get back to it later.
+
+Then insert
+
+```
+<Label fx:id="remark" styleClass="cell_small_label" text="\$remark" />
+```
+into link:https://github.com/j-lum/addressbook-level35/commit/06f9897d7088d22689de478c94612f8bcbff2964#diff-12580431f55d7880578aa4c16f249e71[`main/resources/view/PersonListCard.fxml`].
+
+That's it! Fire up the application again and you should see something like this:
+
+.$remark shows up in each entry
+image::$Remark.png[]
+
+== Modify `Person` to support a `Remark` field
+
+Since `PersonCard` displays data from a `Person`, we need to update `Person` to get our `Remark` displayed!
+
+=== Modify `Person`
+
+We change the constructor of `Person` to take a `Remark`.
+We will also need to define new fields and accessors accordingly to store our new addition.
+
+=== Update other usages of `Person`
+
+Unfortunately, a change to `Person` will cause other commands to break, you will have to modify these commands to use the updated `Person`!
+
+TIP: Use the `Find Usages` feature in IntelliJ IDEA on the `Person` class to find these commands.
+
+== Updating Storage
+
+AddressBook stores data by serializing `JsonAdaptedPerson` into `json` with the help of an external library -- Jackson.
+Let's update `JsonAdaptedPerson` to work with our new `Person`!
+
+While the changes to code may be minimal, the test data will have to be updated as well.
+
+Check out link:https://github.com/j-lum/addressbook-level35/commit/78b2a7438a9f9d828b7f09e190ff6130790966c0[this commit] to see what the changes entail.
+
+== Finalizing the UI
+
+Now that we have finalized the `Person` class and its dependencies, we can now bind the `Remark` field to the UI.
+
+Just add link:https://github.com/j-lum/addressbook-level35/commit/2edaf1c52e07e28dc7a49742bc63352813a55af9[this one line of code!]
+[source, java]
+.PersonCard.java
+----
+public PersonCard(Person person, int displayedIndex) {
+    //...
+    remark.setText(person.getRemark().value);
+}
+----
+
+.The remark label is bound properly!
+image::RemarkBound.png[]
+
```

Comment:
No mention to add remark field to `/data/addressbook.json`
If you don't do that you can't init the data for AB3 after making changes.
I'm not sure how much we are to spoonfeed them in this tutorial, but just a headsup.

Reply:
Yeah this one is something that we MUST mention. I rather add the line than to respond to 3am emails.

Reply:
Added a line to ask them to delete the file and have a new compliant one generated. Let's not risk them editing JSON by hand.